calender/calender.go

```diff
@@ -0,0 +1,395 @@
+	}
+	return allSubmissions, nil
+}
+
+func AddCourseToCalender(db *sql.DB, date time.Time, location null.String, online int8, courseId int, repeats bool, repeatDistance int, repeatEnd time.Time) error {
+	tx, err := db.BeginTx(context.Background(), nil)
+	if err != nil {
+		return err
+	}
+
+	newAppoint := &models.Appointment{Date: date, Location: location, Online: online, CourseID: courseId}
+	err = newAppoint.Insert(context.Background(), tx, boil.Infer())
+	if err != nil {
+		if e := tx.Rollback(); e != nil {
+			return fmt.Errorf("fatal: unable to rollback transaction on error: %s; %s", err, e)
+		}
+		return err
+	}
+	course, err := models.FindCourse(context.Background(), tx, courseId)
+	if err != nil {
+		if e := tx.Rollback(); e != nil {
+			return fmt.Errorf("fatal: unable to rollback transaction on error: %s; %s", err, e)
+		}
+		return err
+	}
+	course.AddAppointments(context.Background(), tx, true, newAppoint)
+
+	var nextDate time.Time = date
+	var checkDate time.Time
+	if repeats {
+		// Go through the calendar at the given interval, insert an appointment; Stop, when the end date is reached
+		for !nextDate.After(repeatEnd) {
+			newAppointAfter := &models.Appointment{Date: nextDate, Location: location, Online: online, CourseID: courseId}
+			err = newAppointAfter.Insert(context.Background(), tx, boil.Infer())
+			if err != nil {
+				if e := tx.Rollback(); e != nil {
+					return fmt.Errorf("fatal: unable to rollback transaction on error: %s; %s", err, e)
+				}
+				return err
+			}
+			course.AddAppointments(context.Background(), tx, true, newAppointAfter)
+
+			// go to next appointment
+			switch repeatDistance {
+			case 1:
+				checkDate = nextDate.AddDate(0, 0, 7) // add seven days
+			case 2:
+				checkDate = nextDate.AddDate(0, 1, 0) // add one month
+			case 3:
+				checkDate = nextDate.AddDate(1, 0, 0) // add a year
+			default:
+				if e := tx.Rollback(); e != nil {
+					return fmt.Errorf("fatal: unable to rollback transaction on error: %s; %s", err, e)
+				}
+			}
+
+			if checkDate.Before(date) {
+				return fmt.Errorf("error when trying to create new appointments")
+			}
+		}
+	} else {
+		course, err := models.FindCourse(context.Background(), tx, courseId)
+		if err != nil {
+			if e := tx.Rollback(); e != nil {
+				return fmt.Errorf("fatal: unable to rollback transaction on error: %s; %s", err, e)
+			}
+			return err
+		}
+		course.AddAppointments(context.Background(), tx, true, newAppoint)
+	}
+
+	if e := tx.Commit(); e != nil {
+		return fmt.Errorf("fatal: unable to Commit transaction on error: %s; %s", err.Error(), e.Error())
+	}
+	return nil
+}
+
+func DeactivateCourseInCalender(db *sql.DB, appointmentId int, courseId int, repeats bool) error {
+	tx, err := db.BeginTx(context.Background(), nil)
+	if err != nil {
+		return err
+	}
+
+	// TODO
+	// repeats: if repeats: delete it more than once? All repeating appointments have same id -> change?
```

Comment:
Was meinst du damit? Die selbe ID in der Datenbank?

Falls ja, das darf nicht sein. Die ID eines Eintrags muss in einer Tabelle immer eindeutig sein; du wirst dann spätestens beim Testen auf den primary key constraint stoßen...

Reply:
@Witcher01 
Falls es ein wiederholender, z.B. wöchentlicher Termin ist, will man ja, dass alle diese Termine gelöscht werden.
Beim schreiben des Kommentars bin ich davon ausgegangen, dass alle Termine die selbe ID haben, was natürlich gar nicht so ist. Die Frage ist, wie man es stattdessen löst